src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -130,10 +130,16 @@ def __init__(self, feature_extractor: Union["SequenceFeatureExtractor", str], *a
         elif (
             feature_extractor._processor_class
             and feature_extractor._processor_class.endswith("WithLM")
-            and kwargs.get("decoder", None) is not None
         ):
-            self.decoder = kwargs["decoder"]
```

Comment:
Doesn't this line crash already when the `decoder` isn't available ? It should.

Maybe the error actually lies in `src/transformers/pipelines/__init__.py` when attempting to load the `decoder`. At least the warning should be there.

Reply:
I reread the code, and the `pipeline(model="mymodelid")`  seems to properly attempt to load the `decoder` and gracefully failover to `ctc` with a warning. Which was the intended behavior.

Are you talking about using the pipeline directly with something like `AutomaticSpeechRecognition(model=model, tokenizer=tokenizer)` ?

Reply:
@Narsil Yes, I am indeed talking about the case when you pass the parameters yourself

Reply:
Ok, then I am unsure we should add that warning.

`pipeline` is the function where we can hide **a lot** of magic.
but the actual classes like `AutomaticSpeechRecognitionPipeline` should be very "standard" pythonic without any magic.

For instance, you cannot pass it a `str` like `"facebook/wav2vec2-base-960h"` you need to pass the real model. It also does not attempt to prevent you from passing your own very custom object enabling advanced usage.

So how do we enable users using a `model with lm` without the `decoder` and without a warning ? We cannot with your proposal (unless adding a magic flag to suppress the warning, which is clunky).

So currently I would say we should keep the current behavior:
- `pipeline` works out magic, makes the best educated guess we can possibly can with all the magic. Gives out warnings when it cannot create optional things (like `decoder`).
- `AutomaticSpeechRecognitionPipeline` is for users that know a little what they do, we make no attempt  of blocking you. But also you can make mistakes too..

Reply:
@Narsil Ok, that's is definitely an option. Closing this pull request then. Thanks for the review :)

It would however then be nice to document the `decoder` parameter, because I did not include it because it is not documented anywhere.

Reply:
You are entirely correct.

Are you going to start the PR ?

Pipeline arguments are many, hard to not make them overwhelming :D